homeassistant/loader.py

```diff
@@ -63,6 +63,72 @@ def manifest_from_legacy_module(domain: str, module: ModuleType) -> Dict:
     }
 
 
+async def _async_get_custom_components(
+        hass: 'HomeAssistant') -> Dict[str, 'Integration']:
+    """Return list of custom integrations."""
+    try:
+        import custom_components
+    except ImportError:
+        return {}
+
+    def get_sub_directories(paths: List) -> List:
+        """Return all sub directories in a set of paths."""
+        return [
+            entry
+            for path in paths
+            for entry in pathlib.Path(path).iterdir()
+            if entry.is_dir()
+        ]
+
+    dirs = await hass.async_add_executor_job(
+        get_sub_directories, custom_components.__path__)
+
+    integrations = await asyncio.gather(*[
+        hass.async_add_executor_job(
+            Integration.resolve_from_root,
+            hass,
+            custom_components,
+            comp.name)
+        for comp in dirs
+    ])
+
+    return {
+        integration.domain: integration
+        for integration in integrations
+        if integration is not None
+    }
+
+
+async def async_get_custom_components(
+        hass: 'HomeAssistant') -> Dict[str, 'Integration']:
+    """Return cached list of custom integrations."""
+    cache = hass.data.get(DATA_CUSTOM_COMPONENTS)
+    if cache is None:
+        cache = hass.data[DATA_CUSTOM_COMPONENTS] = {}
+        cache['lock'] = asyncio.Lock()
+        cache['data'] = None
+    async with cache['lock']:
+        if cache['data'] is None:
+            cache['data'] = await _async_get_custom_components(hass)
+        return cast(Dict[str, 'Integration'], cache['data'])
+
+
+async def async_get_config_flows(hass: 'HomeAssistant') -> List[str]:
+    """Return cached list of config flows."""
+    from homeassistant.generated.config_flows import FLOWS
+    flows = set()  # type: Set[str]
+    flows.update(FLOWS)
+
+    integrations = await async_get_custom_components(hass)
+    flows.update([
+        integration.domain
+        for integration in integrations.values()
+        if integration.config_flow
+    ])
+
+    return list(sorted(flows))
```

Comment:
let's not sort them, as the translated titles are not always alphabetical (ie `hue` -> "Philips Hue"). Maybe just return the set ?

Reply:
Yea, was thinking the same on returning set. It's used as set elsewhere too.